source/DurationPicker.js

```diff
@@ -0,0 +1,690 @@
+			this.doPickerExpired({name:this.name});
+		},
+
+		/**
+		* @private
+		*/
+		countdownChanged: function () {
+			this.$.hour.set('disabled', this.countdown);
+			this.$.minute.set('disabled', this.countdown);
+			this.$.second.set('disabled' , this.countdown);
+		},
+
+		/**
+		* @private
+		*/
+		startTapped: function () {
+			if (this.pauseDisabled) {
+				if(this.values['hour'] || this.values['minute'] || this.values['second']) {
+					this.set('startContent', cancelText);
+					this.startPicker();
+				}
+			} else {
+				this.set('value', '');
+			}
+		},
+
+		/**
+		* @private
+		*/
+		pauseTapped: function () {
+			if (this.countdown) {
+				this.set('pauseContent', resumeText);
+				this.pauseTimer();
+			} else {
+				this.set('pauseContent', pauseText);
+				this.startTimer();
+			}
+		},
+
+		/**
+		* @private
+		*/
+		startPicker: function () {
+			this.set('pauseDisabled', false);
+			this.startTimer();
+		},
+
+		/**
+		* @private
+		*/
+		cancelPicker: function () {
+			this.set('startContent', startText);
+			this.set('pauseContent', pauseText);
+			this.set('pauseDisabled', true);
+			this.pauseTimer();
+		},
+
+		/**
+		* @private
+		*/
+		startTimer: function () {
+			if (this.value) {
+				this.timer = moon.setInterval(enyo.bind (this, function () {
+					this.tick();
+				}), 1000);
+				this.set('countdown', true);
+			}
+		},
+
+		/**
+		* @private
+		*/
+		pauseTimer: function () {
+			this.set('countdown', false);
+			moon.clearInterval(this.timer);
+		},
+
+		/**
+		* @private
+		*/
+		hidePickers: function () {
+			this.set('showHour', false);
+			this.set('showMinute', false);
+			this.set('showSecond', false);
+		},
+
+		/**
+		* @private
+		*/
+		resetPicker: function () {
+			if (!this.pauseDisabled) {
+				this.cancelPicker();
+			}
+			this.values['hour'] = 0;
+			if (this.$.hour) this.$.hour.set('value', 0);
+
+			this.values['minute'] = 0;
+			if (this.$.minute) this.$.minute.set('value', 0);
+
+			this.values['second'] = 0;
+			if (this.$.second) this.$.second.set('value',0);
+		},
+
+		/**
+		* @private
+		*/
+		closePicker: function (sender, ev) {
+			/**
+			* If select/enter is pressed on any duration picker item close the drawer
+			*/
+			if (ev.type == 'onSpotlightSelect' ||
+				this.$.client.children[0].id == ev.originator.id) {
+				this.expandContract();
+				return true;
+			}
+		},
+
+		/**
+		* @private
+		*/
+		toggleActive: function () {
+			if (this.get('open')) {
+				this.set('active', false);
+				if (!enyo.Spotlight.getPointerMode()) {
+					enyo.Spotlight.spot(this.$.headerWrapper);
+				}
+			} else {
+				this.set('active', true);
+			}
+		}
+	});
+
+	(function () {
+		/* A replacement for window.setInterval() function, It use window.setTimeout() to perform the interval functionality
+		* It is based on enyo.perfNow() and use setInterval() and clearInterval() function as like with window
+		*
+		*/
+
+		/* Interval Object */
+		var Interval = function (func, delay) {
```

Comment:
Hi @aarontam, as i mentioned earlier the window.setInterval() function always had +/- 1 ms difference in each interval (1 s). By getting added this in every interval, this will considerably get wrong duration on long run. In current the moon.interval keep the difference consistent from the starting time by using {{enyo.perfNow}}.